File: app/templates/app-java/src/main/java/di/modules/AndroidModule.java
package <%= appPackage %>.di.modules;

import <%= appPackage %>.di.ForApplication;

import android.app.ActivityManager;
import android.app.Application;
import android.app.NotificationManager;
import android.content.Context;
import android.content.SharedPreferences;
import android.view.LayoutInflater;
import android.view.inputmethod.InputMethodManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
@Module
public class AndroidModule {

    @Provides
    public SharedPreferences provideSharedPreferences(@ForApplication Context application) {
        return application.getSharedPreferences(application.getPackageName(), Context.MODE_PRIVATE);
    }

    @Provides
    public LayoutInflater provideLayoutInflater(@ForApplication Application application) {
        return (LayoutInflater) application.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Provides
    public InputMethodManager provideInputMethodManager(@ForApplication Application application) {
        return (InputMethodManager) application.getSystemService(Context.INPUT_METHOD_SERVICE);
    }

    @Provides
    public NotificationManager provideNotificationManager(@ForApplication Context context) {
        return (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
    }

    @Provides
    public ActivityManager provideActivityManager(@ForApplication Application application) {
        return (ActivityManager) application.getSystemService(Context.ACTIVITY_SERVICE);
    }
}

File: app/templates/app-kotlin/src/main/java/di/modules/AndroidModule.kt
package <%= appPackage %>.di.modules

import android.app.ActivityManager
import android.app.Application
import android.app.NotificationManager
import android.content.Context
import android.content.SharedPreferences
import android.view.LayoutInflater
import android.view.inputmethod.InputMethodManager
import dagger.Module
import dagger.Provides
import <%= appPackage %>.di.ForApplication

@Module
class AndroidModule {

    @Provides
    fun provideSharedPreferences(@ForApplication application: Context): SharedPreferences {
        return application.getSharedPreferences(application.packageName, Context.MODE_PRIVATE)
    }

    @Provides
    fun provideLayoutInflater(@ForApplication application: Application): LayoutInflater {
        return application.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    @Provides
    fun provideInputMethodManager(@ForApplication application: Application): InputMethodManager {
        return application.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    }

    @Provides
    fun provideNotificationManager(@ForApplication context: Context): NotificationManager {
        return context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
    }

    @Provides
    fun provideActivityManager(@ForApplication application: Application): ActivityManager {
        return application.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
    }
}
